### PR TITLE
Build nginx image from official image

### PR DESCRIPTION
**Pull request by adammcmaster**
This ditches the old Ubuntu-based build in favour of using the official Docker image, with our preferred default Nginx config (based on the config file in `zooniverse/apps-nginx`). The idea is to streamline our setup; we no longer need the apps-nginx image, and switching to this also removes supervisor and certbot making the images much smaller.

This change won't affect existing apps, because the old images will still be on the registry (I've made sure the automated builds are disabled in Docker Hub so they won't get overwritten).

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    error_log /dev/stderr;
+
+    gzip on;
+    gzip_disable "msie6";
+    gzip_proxied any;
+
+    real_ip_header X-Forwarded-For;
+    real_ip_recursive on;
+    set_real_ip_from 10.0.0.0/8;
+    set_real_ip_from 172.30.0.0/16;
```
Comment (camallen) [inline nginx.conf:31]: what address ranges are these trusted proxies ?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  options {
+    disableConcurrentBuilds()
+  }
+
+  stages {
+    stage('Build Docker image') {
+      agent any
+      steps {
+        script {
+          def nginxVersion = '1.19'
```
Comment (camallen) [inline Jenkinsfile]: So we opt-in to version upgrades by manually updating this?

I'm wondering how we will remember to update these images to release to the next version...

**Review by camallen — APPROVED** [review]
This all LGTM - i assume manually testing the build image artifact will be the best way to test.

File: nginx.conf (hunk 1 above)
Comment (adammcmaster) [reply nginx.conf:31]: These are two private IP ranges. The first is the internal EC2/Azure network and the second is used by Docker if I remember correctly.

File: Jenkinsfile (hunk 2 above)
Comment (adammcmaster) [reply Jenkinsfile]: Yeah I'm not sure how we'll remember that either tbh. We'll need to keep this updated and then also keep the various deployment templates for apps updated.

What we really need is some way to hook that all up to Dependabot or something.

**Pull request merged**